### PR TITLE
More Survival Tools Series 4.5

### PR DESCRIPTION
**Pull request by chaosvolt**
Because I royally fucked up Series 4. Scratch that, I did a thing
suggested by Rivet that became RETROACTIVELY fucked.

Major Changes:
1. Highly belatedly took advantage of the tweaks to use actions, so now
the ember carrier doesn't need its own custom function. Lighting items
with it also takes a turn instead of being instant.
2. Re-did the removal of the TINDERBOX_LIT function.

Item Tweaks and Additions:
1. Work gloves now have 20 encumbrance instead of 10.
2. Tarps. Currently only used as rainproof blankets, tarp tents will be
on another PR.
3. Bone alternative to fishing hooks.

Recipe Tweaks:
1. Recipe for hooded hat can use a bandana for the hood.
2. More tinder options.

Terrain Tweaks:
1. Minor tweaks to the pine lean-to.

Tileset Changes:
1. Tileset additions for a few more items. Tarp tents have sprites that
will go unused for now, they'll exist eventually.

### CODE REVIEW

**Comment by graysage1** [comment]
>You want it gone, get someone more competent to do it.

This is hilarious.  Your competency level isn't something static.  It's something that can develop with time and practice.

**Comment by chaosvolt** [comment]
True, but at the moment I'm not sure I'd trust myself to fix it if it goes wrong. You know, again. >w>

**Comment by graysage1** [comment]
>True, but at the moment I'm not sure I'd trust myself to fix it if it goes wrong. You know, again. >w>

It's a good thing that others don't feel that way, or they'd give up trying to fix things that they've broken repeatedly, like 5 or more times in a row.

**Comment by chaosvolt** [comment]
The thing is that what I did initially wasn't what broke it. It was that a later update to the source caused a merge conflict.

It's annoying when I fail at fixing something, and frustrating when the problem is something I didn't cause. =w=

**Comment by Coolthulhu** [comment]
It's not hard, you just had bad luck there and your changes coincided with mine, which caused the conflict.

Conflict resolution is a valuable skill, though not necessary as long as you're not adding new functions in C++. It is quite easy when you know what to do:
First (only once): git remote add upstream https://github.com/CleverRaven/Cataclysm-DDA.git
Then every time your branch breaks: git checkout master, git pull upstream master, git checkout my-branch, git merge master, git mergetool

The only thing that could change in this command chain is the name of your branch. You don't even need to understand what you're doing when you've got a ready list like that.

**Comment by chaosvolt** [comment]
Hmm. I had tried upstreaming though, at that point it became stuck at the "can't, got a lot of uncommited source files and two conflicts" stages. It also showed that I tried and, it seems, fucked up merging to fix the branch.

I should've grabbed a screenshot. >.<

File: data/mods/More_Survival_Tools/items.json [hunk 1]
```diff
@@ -99,7 +99,20 @@
     "turns_per_charge": 600,
     "ammo": "tinder",
     "revert_to": "tinderbox",
-    "use_action": "TINDERBOX_LIT"
```
Comment (Rivet-the-Zombie) [inline data/mods/More_Survival_Tools/items.json]: Okay now that you've got a nice clean, updated branch please remove the iuse and related entries for TINDERBOX_LIT.

**Comment by chaosvolt** [comment]
Ack. Alright, just hoping I don't wind up hitting bad timing. You know, again.

**Comment by Coolthulhu** [comment]
No risk here. That was a major change (literally rewrite of 80% of the file).

**Comment by chaosvolt** [comment]
Ah, yay. Also, I only recently caught the fact that my original change to player.cpp referred to a non-existent "tinderbox_lit" item instead of "tinderbox_on" in the "does the player have a fire source" check.

I think I'm fine with only being able to light terrain with it. While using a lit ember carrier to light ANOTHER ember carrier would be realistic yet redundant, I can't think of any other realistic use for the "has fire" check. Do we *need* nomads lighting dynamite with little tins of smoldering tinder? o3o

**Comment by graysage1** [comment]
>I think I'm fine with only being able to light terrain with it. While using a lit ember carrier to light ANOTHER ember carrier would be realistic yet redundant, I can't think of any other realistic use for the "has fire" check. Do we need nomads lighting dynamite with little tins of smoldering tinder? o3o

If the ember carrier is not a generic lighter, you should explain that in its description then to avoid confusing the players.

Also, remember how you agreed the volume of the ember carrier should be reduced to 1?  Did you change your mind, or did I just misunderstand you?

**Comment by chaosvolt** [comment]
Hmm, will have to check the description to see if its use for starting terrain fires is clear enough.

And...I actually forgot. On it. Fucking net, stop derping and not showing me replies. D:

**Comment by Coolthulhu** [comment]
* Pine lean-to looks just like a shrub in ASCII mode
* Tarp "armor" is 6/6 at 100% coverage. While the 3 encumbrance makes it not too good of an armor, the 100% coverage with 6/6 protection makes it not bad. Drop some of that coverage.
* What's the deal with 3 almost identical tarp lean-to recipes? Merge them somehow. Make them always require and return tarp and make tarp armor craftable from blankets and ponchos and forbid reversing the recipe.
* Remove the pebbles from tarp lean-to. They're a meaningless detail.
* You could replace spikes with spiked spiked sticks. Someone with an access to lots of spikes will most likely not need to construct a lean-to.
* Bone fishing hook can't be used. The `"sub"` option only works for tools and the hook is a component (ie. it is consumed by the recipe).
I didn't test the tileset.

The rest looks fine.

**Comment by chaosvolt** [comment]
Ack. I think I took the tarp's coverage from the sheet or blanket stats, so I can tone down the coverage and material thickness. As for merging the terrain, I guess that'd work. And would make the pebbles unnecessary.

Should I just be a dunce and give such a recipe a cutting quality, or add some sort of item to function as grommets? >w>

**Comment by chaosvolt** [comment]
Meep. Here's hoping I haven't fucked anything else up. ;w;

File: data/mods/More_Survival_Tools/items.json [hunk 2]
```diff
@@ -84,12 +84,12 @@
     "color": "light_gray",
     "name": "ember carrier (lit)",
     "name_plural": "ember carriers (lit)",
-    "description": "This is a small container for holding tinder, with holes punched in it to provide a controlled airflow.  An ember is smoldering inside it, and can be used to light a fire.",
+    "description": "This is a small container for holding tinder, with holes punched in it to provide a controlled airflow.  An ember is smoldering inside it, and can be used to start a campfire.",
     "price": 1000,
     "material": "iron",
     "flags": ["FIRE", "FLAMING", "REFILLABLE_LIGHTER"],
```
Comment (Coolthulhu) [inline data/mods/More_Survival_Tools/items.json]: Minor: None of those flags matter. Only "FLAMING" does anything and what it does is dealing fire damage on melee hit.

**Comment by chaosvolt** [comment]
Ah derp. Pulling the "FLAMING" flag can be done on a later PR perhaps. owo

**Comment by Coolthulhu** [comment]
Sure, those are just minor things.

But once again: next time push tileset changes as a separate PR. It's OK for some symbols (especially just equipment) to be '?' for a while.

**Comment by chaosvolt** [comment]
Meep. Understood, though the main thing that prompted me to push tileset tweaks was the addition of tarp tents. @_@

**Comment by kevingranade** [comment]
If that's the case, consider breaking tarp tents and their tiles into a separate PR.

**Comment by chaosvolt** [comment]
Ack. x.x

**Comment by kevingranade** [comment]
I mean consider it in the future, looks like Coolthuku is in the process of merging it, so it'll be unecessary now.
Also all we mean is if you break it up more it'll likely be merged faster.

**Comment by Coolthulhu** [comment]
Minor:
* Constructing a pine lean-to turns a tree into a passable tile.
* Pine lean-to uses a green background, which we currently use for bushes and thus should not be used for . Not sure what would be a good symbol for it, though. Maybe `;`? It is used for animalskin floor.
* Pine lean-to turns an unpassable terrain passable, then back when deconstructed. Upon deconstruction the items stay "wedged" inside the pine.
* Tarp has heavy requirements - 3 farbication and 2 survival. It would be fine with 1 of each. It also has weird cost - 8 duct tape and 4 scrap metal. Should be about 20 duct tape and no metal.

File: gfx/MShock24TilesetModded/tile_config.json [hunk 3]
```diff
@@ -19278,6 +19278,37 @@
...
+        {
+          "id": "fishing_rod_professional",
+          "fg": 2516,
+          "rotates": false
+        },
+        {
+          "id": "fishing_hook_basic",
+          "fg": 2517,
+          "rotates": false
+        }
```
Comment (Coolthulhu) [inline gfx/MShock24TilesetModded/tile_config.json]: "Failed to load tileset: missing separator"

**Comment by chaosvolt** [comment]
Hnng. Ack. As for the pine lean-to's color, I thought I took it from the improvised shelter. As for the passable problem...hmm. You can't deconstruct improvised shelters either, so I might simply remove that option.

And...oh crap. Rather than yoink the tileset changes to separate them now, I think I'll fix it and keep it in mind for the next PR. Speaking of which, I see I already have something to fix. ;w;

**Comment by chaosvolt** [comment]
Also, a derp idea. A terrain item that works as a refinement of the "containing a fire in a pit" idea, a Dakota fire hole. Would have to see if terrain can both contain fire AND smoke.

If a terrain can also affect how much light a fire inside it gives off, that would also be interesting. No idea how to go about any such thing though. owo

**Comment by Coolthulhu** [comment]
Fire pit currently won't work due to hardcoding. It checks for `t_pit`, not `"FIRE_PIT"` flag or anything like that.

Fire and smoke can exist in the same tile, just aren't displayed properly.

**Comment by chaosvolt** [comment]
Hmm. Right, so basic pits do at least prevent the spread of fire?

I was thinking a furniture item with the fire and smoke suppression tokens would work, but that would potentially be redundant if pits are already viable for fire suppression, as smoke suppression wouldn't be as important since it's assumed you're building it outdoors.

**Comment by chaosvolt** [comment]
Whee. I think I'll need another sip of mead for this. owo

**Comment by chaosvolt** [comment]
And thank you for the merge. <3

**Comment by chaosvolt** [comment]
Also, no notes for the next PR? I'm surprised I've actually reached that point. owo

**Comment by kevingranade** [comment]
Pits do contain fires.

**Comment by chaosvolt** [comment]
As I've mentioned finding out the hard way, shallow ones at least don't seem to. owo

**Comment by graysage1** [comment]
@chaosvolt Is it possible to close the dialog right after the selection which asks whether to light a fire quickly or extinguish when using ember carrier?

Right now when I select "Light fire quickly" the dialog stays open until I also pick a direction, which I don't think makes sense.  Also the dialogue somewhat obscures the place where I want to start a fire, so it's good to close it right away after the selection, imo.

**Comment by Coolthulhu** [comment]
@graysage1 From mod side, no. But from the code side it should be just a single draw call.
Until then, you can use the item action menu. As far as I recall, ember carrier uses 0 charges, so it should be picked as default for the item action menu since it doesn't understand things like move cost.

**Comment by chaosvolt** [comment]
Aw crap. NOW I see a use for the TINDERBOX_LIT function that's since been removed. Fuck. ._.

**Comment by chaosvolt** [comment]
Aw crap. Didn't realize that would occur. ;w;

**Comment by graysage1** [comment]
>From mod side, no.

I see.

>I'm guessing the old TINDERBOX_LIT function had the same issue, or...?

I don't know.  Maybe I just never noticed it before?

**Comment by chaosvolt** [comment]
I'm going to guess that TINDERBOX_LIT worked the same way. If it didn't, I'm gonna be pissed at that now-removed function suddenly becoming useful. ._.

**Comment by graysage1** [comment]
@chaosvolt I am an idiot.  I completely forgot I have a significant history of past builds and their associated save files here in my little "archive."  I just pulled up 0.C-3035 and took a look.

I notice that the popup for the ember carrier was smaller, it had "light something" as a second choice instead of currently first choice (which was worse), **but**, the popup did used to close as soon as the choice was made.  It left a dark unpainted background behind it, which is a slight bug, but the point it, the popup for the ember carrier did used to behave closer to how I liked it than currently.

**Comment by chaosvolt** [comment]
Hnng. Well crap, not sure whether to restore that or not. Might be worth testing whether or not that would work the same way in the current version. Wouldn't know if it was my copy-pastes of TORCH_LIT that made it work, or if the underlying code has been changed in these later versions.

Wait. Doh. Test how the menu works when using a lit torch, that's where I copy-pasted the source when making the TINDERBOX_LIT function.

**Comment by graysage1** [comment]
>Wouldn't know if it was my copy-pastes of TORCH_LIT that made it work, or if the underlying code has been changed in these later versions.

I'm afraid you'll have to discuss this with people who actually know the codebase.  I am not one of them. :)  I can only help you with some testing.  If you want me to test something, I can do that.

>Test how the menu works when using a lit torch

OK.  In 0.C-3035 lit torch seems to be identical to how lit ember carrier works.  I'll test the latest now.

In 0.C-3106, the torch works as before, but the ember carrier is now different.  Firstly the option to light a fire is first on the list, which is good.  But it also has a much longer verbiage, so the popup is bigger, *and*, most importantly, the popup doesn't close after you hit enter on the first option, but stays open until you also select a direction for the fire.

**Comment by chaosvolt** [comment]
Hmm. So the popup disappears when choosing to light with the torch, in the current version?

Crap. >.<

**Comment by Coolthulhu** [comment]
You could fix it by adding a `g->draw_ter` in `firestarter_actor::prep_firestarter_use`, just before `choose_adjacent`.

**Comment by graysage1** [comment]
>So the popup disappears when choosing to light with the torch, in the current version?

I'm 99% sure yes.

**Comment by chaosvolt** [comment]
Hmm. Crap. dunno what to do. >.<

**Pull request merged**